src/lib.rs

```diff
@@ -131,19 +139,12 @@ impl<T: Runtime> ClientBuilder<T> {
     pub fn new() -> Self {
         Self {
             url: None,
-            client: None,
             page_size: None,
             event_type_registry: EventTypeRegistry::new(),
             skip_type_sizes_check: false,
         }
     }
 
-    /// Sets the jsonrpsee client.
-    pub fn set_client<P: Into<jsonrpsee::Client>>(mut self, client: P) -> Self {
```

Comment:
Probably better to split this into `set_http_client` and `set_ws_client` or so. We currently set the client this way since we require access for custom rpc calls.

Reply:
Additionally I see no reason why we couldn't pass a mock client that implements [these traits](https://github.com/paritytech/jsonrpsee/blob/master/types/src/traits.rs), or am I missing something?

Reply:
Good questions, "you could" but the problem is how the `proc macros` for frame/Call/Storage is generated I wrote a few lines about it in the PR description.

Basically you would need to change to:
```rust
struct ClientBuilder<T: Runtime, Rpc: jsonrpsee::traits::Client> {
 client: Option<Rpc>,
 ....
} 
```

This type param is then leaked into lower layers and it would require major refactoring which I prefer not do in this PR. Another option is to use traits object `Box<dyn jsonrpsee::traits::Client>` and that won't work because the trait itself is not object safe.

Thoughts? :)

Reply:
@niklasad1 perhaps we can adopt the following approach: https://github.com/paritytech/substrate-subxt/pull/236

Reply:
What do you think? I would be happy to finalize the update if you think it is a viable workaround.

Reply:
Commented in the PR but I don't think it will work "out-of-the-box", I will try it out.

Reply:
Replied and followed up, took me most day but the embedded client should now work...